File: code/ex287.java
class Solution {
    public int findDuplicate(int[] nums) {
        int[] record = new int[nums.length];

        for (int i = 0; i < nums.length; i++) {
            if (record[nums[i] - 1] == 1)   return nums[i];
            record[nums[i] - 1]++;
        }

        return 0;
    }
}

File: code/ex287.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int fast = nums[nums[0]], slow = nums[0];
        while (fast != slow) {
            fast = nums[nums[fast]];
            slow = nums[slow];
        }
        
        slow = 0;
        while (slow != fast) {
            slow = nums[slow];
            fast = nums[fast];
        }
        return slow;
    }
};